containertest/mysql.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package containertest
+
+import (
+	"database/sql"
+	"fmt"
+	"net"
+
+	_ "github.com/go-sql-driver/mysql" // Force mysql service to be included.
+)
+
+const (
+	// It's OK to hardcode the root password because only boilerplate test data is stored. Also,
+	// having a well-known password can help with human inspection for debugging. The value chosen for
+	// the password is arbitrary. It can be changed without breaking anything; it's not hardcoded into
+	// the docker image or anything like that.
+	password = "8mo5lfYKjy6ebTK" //nolint:gosec
```

Comment:
This can be a follow-up, but since we're abstracting this now, I'd like to generate a random password instead.

Reply:
``` go
func generatePassword() string {
	return "a random password"
}
```

Sure, I can do. I'll probably just generate a short alphanumeric. Special characters will mess up my uri formatting because I didn't escape the uri.

Reply:
Though, there is a counter argument:

It could be nice to be able to put a breakpoint in your test code and attach a db client, which would be more difficult if the password changes around

Reply:
I don't know if having a function return a random password works, since you'd want it to be deterministic for each instance. Therefore I think you'd need a password field on the structure and then on creation you'd generate a random password. You can avoid the escaping need by cheating a little bit and reading ~16 random bytes and then hex encoding it (which gives you 32 characters).

I agree on the counterargument. My counter-counter arguments are:

- We could print out the password (still kinda sucks)
- You could configure container auth to allow all local network traffic (and therefore no password is required)

Reply:
I think I'll leave as-is for now as none of the proposed solutions would require an interface change. Let me know if you want me to add an issue.